packages/neon_framework/lib/src/utils/user_agent.dart

```diff
@@ -1,27 +1,20 @@
 import 'package:meta/meta.dart';
 import 'package:package_info_plus/package_info_plus.dart';
 
-String? _userAgent;
-
-/// Sets the user agent.
+/// Builds the user agent for this neon application.
 ///
-/// It can be accessed with [neonUserAgent].
+/// If the [appName] is `'Nextcloud Neon'` it will be omitted.
 @internal
-void buildUserAgent(PackageInfo packageInfo) {
+String buildUserAgent(PackageInfo packageInfo, [String? appName]) {
+  final buffer = StringBuffer('Neon ');
+  if (appName != null && appName != 'Nextcloud Neon') {
```

Comment:
Shouldn't this check against the localizations?
Also this would be easier by just not passing the appName if it is `Nextcloud Neon`.
Not sure where this is called from in that case though.

Reply:
We need to have the magic string somewhere, as the `Nextcloud Neon` case is only defined in the example/app package and not in the framework itself.

I did think of making it the fallback value in the `Branding` class, but I fear that other devs might not set it because it already has a convenient value.
Long term, there should only be a single call to that function and every communication should go through a single http client. I'll leave it as is for now.